### PR TITLE
Text filtered hints

### PR DESCRIPTION
**Pull request by vlmarek**
Hi,

So this is where I stand now. I'm happy with the code, but I gave it just a little testing so far. Since everything is new for me I don't expect to have very polished code.

I still miss regular expressions and searching on a whole page, not just in the visible part. I wold like to implement find patters similar to what pentadactyl had

http://bug.5digits.org/help/pentadactyl/pattern.xhtm

\c    Perform case insensitive find
\C    Perform case sensitive find
\L    Search the entire page.
\r    Process the entire pattern as a regular expression.
\R    Process the entire pattern as an ordinary string.

Thank you for everyone's support

### CODE REVIEW

**Comment by vlmarek** [comment]
Hi,

I have updated the changes to be lint clean. I had to rewrite the hintPage function even more, but I believe that it is better now

File: src/content/hinting.ts [hunk 1]
```diff
@@ -716,7 +716,11 @@ export function hintByText(match) {
             } else {
                 text = hint.textContent
             }
-            return text.indexOf(match) >= 0
+            if (match.constructor.name == "RegExp") {
+                return text.match(match) !== null
+            } else
+                return text.indexOf(match) >= 0
+            }
```
Comment (cmcaine) [inline src/content/hinting.ts]: use `text.search(match) >= 0` instead. It works for both regexp and string patterns.

**Comment by cmcaine** [comment]
You probably noticed that I already merged your older hint -f stuff. It didn't appear here on github because I had to rebase. I'll merge once the issue above is fixed (or I'll fix it later and merge anyway ;)

File: src/content/hinting.ts (hunk 1 above)
Comment (vlmarek) [reply src/content/hinting.ts]: I have tried that and it seems that it always interprets the text as a regex. That is
`:hint -f $`
matches all links, not only those containing '$'. This seems to confirm it: https://www.w3schools.com/jsref/jsref_match.asp

Comment (cmcaine) [reply src/content/hinting.ts]: Sorry, my bad there.

Please use `match instanceof RegExp` for the check and `text.includes(match)` for the string match.

Happy to merge then.

Comment (vlmarek) [reply src/content/hinting.ts]: Thanks for the tips, it's the first time I'm touching typescript. I have also added some comment above the function.

**Comment by bovine3dom** [comment]
I made `hint -f` case insensitive because it surprised me when it wasn't. Merged. Thanks!

**Pull request merged**